lib/lambda_ethereum_consensus/store/state_store.ex

```diff
@@ -7,43 +7,54 @@ defmodule LambdaEthereumConsensus.Store.StateStore do
   alias Types.BeaconState
 
   @state_prefix "beacon_state"
+  @state_block_prefix "beacon_state_by_state"
   @stateslot_prefix @state_prefix <> "slot"
 
-  @spec store_state(BeaconState.t(), Types.root()) :: :ok
+  @spec store_state(BeaconState.t()) :: :ok
   def store_state(%BeaconState{} = state) do
     # NOTE: due to how SSZ-hashing works, hash(block) == hash(header)
-    store_state(state, Ssz.hash_tree_root!(state.latest_block_header))
-  end
-
-  def store_state(%BeaconState{} = state, block_root) do
+    block_root = Ssz.hash_tree_root!(state.latest_block_header)
+    state_root = Ssz.hash_tree_root!(state)
     {:ok, encoded_state} = Ssz.to_ssz(state)
 
-    key = state_key(block_root)
-    Db.put(key, encoded_state)
+    key_block = state_key(block_root)
+    key_state = block_key(state_root)
+    Db.put(key_block, encoded_state)
+    Db.put(key_state, key_block)
```

Comment:
We should just store the `block_root` here. That saves on space, and we can go from `block_root` to `block_key` anyways